wal/wal.go

```diff
@@ -532,21 +540,28 @@ func (w *WAL) saveState(s *raftpb.HardState) error {
 }
 
 func (w *WAL) Save(st raftpb.HardState, ents []raftpb.Entry) error {
-	w.mu.Lock()
```

Comment:
saving one allocation will require having a static byte buffer and using the same buffer with invoking MarshalTo(data []byte) ? 

if we use a static byte buffer, we will need the mutex lock always. right?